pyicloud/services/drive.py

```diff
@@ -42,6 +48,8 @@ def get_node_data(self, node_id):
                 ]
             ),
         )
+        if not request.ok:
+            self._raise_error(request.status_code, request.reason)
```

Comment:
Shouldn't we raise an error inside the session `request` method before returning the response? We already do this for some not ok responses.

Reply:
It has been 14 months since I worked on this code but I'm pretty sure `session.request` does not catch all cases where `request.ok` is false. This current code agrees with me:

https://github.com/picklepete/pyicloud/blob/a14a57743d4cb966ee158359b5caa61d190a59ab/pyicloud/services/drive.py#L55-L56

I can easily patch this in `base.py` but I wonder what effect this will have elsewhere?

https://github.com/picklepete/pyicloud/blob/a14a57743d4cb966ee158359b5caa61d190a59ab/pyicloud/base.py#L99-L102

For example `RemindersService` returns the `ok`:

https://github.com/picklepete/pyicloud/blob/a14a57743d4cb966ee158359b5caa61d190a59ab/pyicloud/services/reminders.py#L123

Reply:
Ok. Yeah, it's a bigger change for sure. I don't know any details here. It's safer to just raise in this module until someone wants to take a larger view and perhaps rework this. Let's save that for the future. :+1:

Reply:
Agreed.